File: app/javascript/dashboard/shared/models/article.ts
import { User } from './user'
import { Comment } from './comment'
import { Tag } from './tag'

export class Article {
  id: number
  slug: string
  created_at: string
  updated_at: string
  title: string
  content: string
  favorites_count: number
  comments_count: number
  user: User
  favorite_users: User[]
  comments: Comment[]
  tags: Tag[]
}

File: app/models/article.rb
class Article < ApplicationRecord

  extend FriendlyId
  friendly_id :title

  belongs_to :user
  has_many :comments
  has_many :favorites
  has_many :favorite_users, through: :favorites, source: :user
  has_and_belongs_to_many :tags

  validates :title, presence: true

end
